### ISSUE TITLE
[no-extran-class] `allowWithDecorator` should ignore  other errors

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3114]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```json
{
  "rules": {
    "@typescript-eslint/no-extran-class": [
      2,
      {
        "allowWithDecorator": true
      }
    ]
  }
}
```

```ts
@NgModule({
  // ...
})
export class AppModule {
  constructor(translate: TranslateService) {
    translate.locale$.subscribe(locale => {
      document.documentElement.setAttribute('lang', locale);
    });
  }
}
```

**Expected Result**

Pass

**Actual Result**

```log
error  Unexpected class with only a constructor  @typescript-eslint/no-extraneous-class
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.2` |
| `@typescript-eslint/parser`        | `4.15.2` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `12.20.0` |

**Comment by JounQin** [comment]
cc @a-tarasyuk
